IDEMPIERE-6163 : Adding DisplaySQL  in Reference > Table Validation

dpansheriya opened the pull request:
https://idempiere.atlassian.net/browse/IDEMPIERE-6163

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added support for custom display SQL in reference lookups, allowing more flexible presentation of lookup data.
  * Introduced identifier display configuration for reference tables, enabling users to designate and display specific fields as identifiers in lookup results.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dpansheriya committed:
IDEMPIERE-6163 : Adding DisplaySQL  in Reference > Table Validation

norbertbede commented:
@PeterTakacs300 pls do functional test

sbhimani12 committed:
Merge branch 'master' into IDEMPIERE-6163

dpansheriya committed:
IDEMPIERE-6163: Supporting DisplaySQL and Display Identifier field in InfoGeneralPanel. moving migration script to id13

dpansheriya commented:
Changed migration script directory to iD13 and made fix to support feature in infoGeneralPanel

CarlosRuiz-globalqss reviewed (changes requested):
please check comments at
https://idempiere.atlassian.net/browse/IDEMPIERE-6163?focusedCommentId=51729

dpansheriya committed:
IDEMPIERE-6163: Adding Display SQL in paranthesis, if no displaySQL and if column is _ID then it should show as identifier column

dpansheriya committed:
IDEMPIERE-6163: Making sure display SQL is in text format to avoid NVL function not found error.

dpansheriya commented on org.adempiere.base/src/org/compiere/model/X_AD_Ref_Table.java:
```diff
@@ -31,6 +31,7 @@ public class X_AD_Ref_Table extends PO implements I_AD_Ref_Table, I_Persistent
 	/**
 	 *
 	 */
+
```
This was taken care

dpansheriya commented:
> please check comments at https://idempiere.atlassian.net/browse/IDEMPIERE-6163?focusedCommentId=51729

I did not responded here but pushed this changes on 9th March. so I think there is nothing know pending from my side.

norbertbede commented:
may i ask @hengsin @CarlosRuiz-globalqss a review ? we would like to use it soon.

dpansheriya committed:
Merge branch 'master' into IDEMPIERE-6163

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss committed:
peer review

Removed an unnecessary insert statement and added a new update statement for AD_Field.

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss committed:
peer review

CarlosRuiz-globalqss reviewed (approved):
tested - there is another commit commit for MRefTable
